subsys/bluetooth/mesh/friend.c

```diff
@@ -474,7 +474,7 @@ static int encrypt_friend_pdu(struct bt_mesh_friend *frnd, struct net_buf *buf,
 
 	src = sys_get_be16(&buf->data[5]);
 
-	if (bt_mesh_elem_find(src)) {
+	if (bt_mesh_has_addr(src)) {
```

Comment:
Since `src` is already a unicast address, there is no need to modify it.

Reply:
I agree that this has no functional change. The original use case of bt_mesh_elem_find in this instance returns an unnecessary object, bt_mesh_elem*, which the calling function, in this case encrypt_friend_pdu, doesn't actually make use of. The purpose behind switching to bt_mesh_has_addr is to provide a more sensible return value and improve code readability by more accurately describing the intentions of the code.